src/main/java/org/corfudb/infrastructure/SequencerServer.java

```diff
@@ -166,23 +170,27 @@ public synchronized void handleMessage(CorfuMsg msg, ChannelHandlerContext ctx,
                         max = globalIndex.get() - 1;
                     }
                     r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(
-                                    new TokenResponse(max, Collections.emptyMap(), streamsLastIssued.build())));
+                                    new TokenResponse(max, Collections.emptyMap(), latestStreamOffsets.build())));
                 } else {
                     long thisIssue = globalIndex.getAndAdd(req.getNumTokens());
                     if (req.getStreams() == null) {
                         r.sendResponse(ctx, msg, CorfuMsgType.TOKEN_RES.payloadMsg(
                                 new TokenResponse(thisIssue, Collections.emptyMap(), Collections.emptyMap())));
                         return;
                     }
+                    // If the request is a transaction resolution request, then check if it should abort.
                     if (req.getTxnResolution()) {
-                        // Then also need a read timestamp.
+                        // Read timestamp of the txn; in order to commit the txn, no writes may be made past this
+                        // (global) timestamp on any streams touched by the txn.
                         long timestamp = req.getReadTimestamp();
+                        // If the timestamp is -1L, then the transaction automatically commits.
                         if (timestamp != -1L) {
+                            // This variable is to indicate whether the transaction should abort.
                             AtomicBoolean abort = new AtomicBoolean(false);
                             for (UUID id : req.getStreams()) {
                                 if (abort.get())
                                     break;
-                                lastIssuedMap.compute(id, (k, v) -> {
+                                lastGlobalTokenMap.compute(id, (k, v) -> {
```

Comment:
returning null here seems buggy. This would leave a client hanging if it queries about a stream that the sequencer does not know about. Even if this client behavior is buggy, we should return an error response, not bail out.

Also, I wonder if the following can happen: A client asks to commit on a stream that did not have any updates yet. Would the sequencer have anything in its stream-map about this stream already?